mozjs-sys/build.rs

```diff
@@ -696,434 +556,593 @@ impl ParseCallbacks for CustomCargoCallbacks {
-            .enable_cxx_namespaces()
-            .allowlist_file("./src/jsglue.cpp")
-            .allowlist_recursively(false)
-            .clang_args(["-include", &confdefs_path.to_str().expect("UTF-8")]);
-
-        if msvc {
-            builder = builder.clang_args([
-                "-fms-compatibility",
-                "-DWIN32",
-                "-D_CRT_USE_BUILTIN_OFFSETOF",
-                "-std=c++17",
-            ])
-        } else {
-            builder = builder.clang_args(["-fPIC", "-fno-rtti", "-std=c++17"])
+        pub(crate) fn whitelist_types(self) -> &'static [&'static str] {
+            match self {
+                BuildTarget::JSApi => jsapi::WHITELIST_TYPES,
+                BuildTarget::JSGlue => &[],
+            }
         }
 
-        for ty in BLACKLIST_TYPES {
-            builder = builder.blocklist_type(ty);
+        pub(crate) fn whitelist_vars(self) -> &'static [&'static str] {
+            match self {
+                BuildTarget::JSApi => jsapi::WHITELIST_VARS,
+                BuildTarget::JSGlue => &[],
+            }
+        }
+
+        pub(crate) fn whitelist_functions(self) -> &'static [&'static str] {
+            match self {
+                BuildTarget::JSApi => jsapi::WHITELIST_FUNCTIONS,
+                BuildTarget::JSGlue => &[],
+            }
         }
 
-        for ty in OPAQUE_TYPES {
-            builder = builder.opaque_type(ty);
+        pub(crate) fn blacklist_types(self) -> &'static [&'static str] {
+            match self {
+                BuildTarget::JSApi => jsapi::BLACKLIST_TYPES,
+                BuildTarget::JSGlue => jsglue::BLACKLIST_TYPES,
+            }
         }
 
-        for &(module, raw_line) in MODULE_RAW_LINES {
-            builder = builder.module_raw_line(module, raw_line);
+        pub(crate) fn blacklist_functions(self) -> &'static [&'static str] {
+            match self {
+                BuildTarget::JSApi => jsapi::BLACKLIST_FUNCTIONS,
+                BuildTarget::JSGlue => &[],
+            }
         }
-        let bindings = builder
-            .generate()
-            .expect("Unable to generate bindings to jsglue");
 
-        bindings
-            .write_to_file(outdir.join("gluebindings.rs"))
-            .expect("Couldn't write bindings!");
+        pub(crate) fn opaque_types(self) -> &'static [&'static str] {
+            match self {
+                BuildTarget::JSApi => jsapi::OPAQUE_TYPES,
+                BuildTarget::JSGlue => jsglue::OPAQUE_TYPES,
+            }
+        }
+
+        pub(crate) fn module_raw_lines(self) -> &'static [(&'static str, &'static str)] {
+            match self {
+                BuildTarget::JSApi => jsapi::MODULE_RAW_LINES,
+                BuildTarget::JSGlue => jsglue::MODULE_RAW_LINES,
+            }
+        }
+    }
+
+    mod jsapi {
+        /// JSAPI types for which we should implement `Sync`.
+        pub(crate) const UNSAFE_IMPL_SYNC_TYPES: &'static [&'static str] = &[
```

Comment:
can we just inline uses and remove `mod`s

Reply:
This is possible, but I thought it would look weird if the functions had super long bodies that was just a constant.